### PR TITLE
TEST-#3188: remove Dask pin

### LINKED ISSUE #3188: Remove Dask pin in setup.py
https://github.com/conda-forge/modin-feedstock/pull/17

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [x] Resolves #3188 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
> @anmyachev LGTM, is it ready to move from draft?

No, with the current versions of dask and distributed(2.21.0), some CI jobs usually work much longer (up to 6 hours). Maybe deadlock. I will try another versions.

File: environment-dev.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ dependencies:
   - numpy>=1.16.5
   - pyarrow>=1.0.0
   - libthrift<0.14.1  # ref: modin gh-2840
-  - dask[complete]>=2.12.0,<=2.19.0
-  - distributed>=2.12.0,<=2.19.0
+  - dask[complete]>=2.22.0
```
Comment (devin-petersohn) [inline environment-dev.yml:9]:
```suggestion
  - dask[complete]>=2020
```

**Review by devin-petersohn — COMMENTED** [review]
They changed their version numbers to follow year & month recently. Maybe we can try starting with 2020.

It might be good to put a timeout on the jobs too so they don't eat our minutes.

File: environment-dev.yml (hunk 1 above)
Comment (vnlitvinov) [reply environment-dev.yml:9]: As we've removed the upper pin, it now installs whatever is recent enough, doesn't it? I see `dask 2021.6.2` being used right now in CI.

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Comment by anmyachev** [comment]
> They changed their version numbers to follow year & month recently. Maybe we can try starting with 2020.
> 
> It might be good to put a timeout on the jobs too so they don't eat our minutes.

`2.22.0` version works, so we can leave it.
We can choose not to set timeouts until the deadlock occurs frequently.

**Pull request merged**